crates/bevy_input/src/gamepad.rs

```diff
@@ -1213,39 +1216,138 @@ impl AxisSettings {
     }
 
     /// Clamps the `raw_value` according to the `AxisSettings`.
-    pub fn clamp(&self, new_value: f32) -> f32 {
-        if self.deadzone_lowerbound <= new_value && new_value <= self.deadzone_upperbound {
+    pub fn clamp(&self, raw_value: f32) -> f32 {
+        if self.deadzone_lowerbound <= raw_value && raw_value <= self.deadzone_upperbound {
             0.0
-        } else if new_value >= self.livezone_upperbound {
+        } else if raw_value >= self.livezone_upperbound {
             1.0
-        } else if new_value <= self.livezone_lowerbound {
+        } else if raw_value <= self.livezone_lowerbound {
             -1.0
         } else {
-            new_value
+            raw_value
         }
     }
 
-    /// Determines whether the change from `old_value` to `new_value` should
+    /// Determines whether the change from `old_raw_value` to `new_raw_value` should
     /// be registered as a change, according to the [`AxisSettings`].
-    fn should_register_change(&self, new_value: f32, old_value: Option<f32>) -> bool {
-        if old_value.is_none() {
-            return true;
+    #[inline(always)]
```

Comment:
Never inline private methods. Same for others. [Here](https://matklad.github.io/2021/07/09/inline-in-rust.html) is a good article about it.
```suggestion
```